### PR TITLE
Replaced random thread information with actual youtube video title & sentiment score

### PR DESCRIPTION
**Pull request by shivanighanta**
- Replaced random thread title with actual youtube video title
- Replaced random sentiment score with sentiment score of the content of a youtube video
- Added getters in YoutubePost.java
image

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -68,19 +67,20 @@
...
     try {
-      List<YoutubePost> newPost = YoutubeApi.getYoutubePost();
+      newPost = YoutubeApi.getYoutubePost();
     } catch (YoutubeApiException e) {
       System.out.println("Error: Youtube api returning exception" + e);
       response.sendError(500, "An error occurred while fetching Youtube Posts");
       return;
     }
     if (threadTitles.size() <= 0) {
       for (int i = 0; i < 5; i++) {
```
Comment (jonaner) [inline src/main/java/com/google/sps/servlets/DataServlet.java]: Can you turn this to a for each loop just to make sure we get all of the posts?

**Review by BlessP — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -44,13 +44,12 @@
...
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import javax.xml.ws.http.HTTPException;
 import org.json.simple.JSONObject;
 
 /**
  * Servlet responsible for storing Youtube Video and Displaying the details of the Youtube Video
  */
-@WebServlet("/data")
+@WebServlet("/data1")
```
Comment (jbiral) [inline src/main/java/com/google/sps/servlets/DataServlet.java]: nit: can you revert this back to `/data`?

If the endpoint is stale, try to delete the `targets/` folder, or feel free to reach out to VSEs to debug further, but I don't think this is the long term solution, so we shouldn't be committing this.

However, if you want to rename the endpoint, I think it is a good idea. Can we pick a more descriptive name? For example, `/videos-sentiment`?

File: src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (jbiral) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: +1, great suggestion. It's fine if we just have one post for now, we will retrieve multiple posts in a follow up

File: src/main/java/com/google/sps/servlets/YoutubePost.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
   private final String title;
   private final String content;
   private static String url;
-  private final String id;
+  private static String id;
```
Comment (jbiral) [inline src/main/java/com/google/sps/servlets/YoutubePost.java]: Actually, these shouldn't be static. Can we make them final?

File: src/main/java/com/google/sps/servlets/YoutubePost.java [hunk 4]
```diff
@@ -34,6 +34,18 @@ public static String getUrl(String id) {
     return "https://www.youtube.com/watch?v=" + id;
   }
 
+  public static String getID() {
+    return id;
+  }
```
Comment (jbiral) [inline src/main/java/com/google/sps/servlets/YoutubePost.java]: Please make this non-static

File: src/main/java/com/google/sps/servlets/YoutubePost.java [hunk 5]
```diff
@@ -34,6 +34,18 @@ public static String getUrl(String id) {
     return "https://www.youtube.com/watch?v=" + id;
```
Comment (jbiral) [inline src/main/java/com/google/sps/servlets/YoutubePost.java]: Please make this non-static as well, and ideally it'd be great if the getter just return `this.url`, but the formatting is done on line 30.

File: src/main/java/com/google/sps/servlets/AnalyzedVideo.java [hunk 6]
```diff
@@ -61,8 +63,9 @@ public static String getRandomTitle() {
     return titlesList.get(rand.nextInt(4));
   }
 
-  public static double getRandomSentiment() {
-    return sentimentList.get(rand.nextInt(4));
+  public static double getSentiment(String text) throws IOException {
+    Sentiment sentimentFromText = analyze.analyzeSentimentText(text);
+    return sentimentFromText.getScore();
   }
```
Comment (jbiral) [inline src/main/java/com/google/sps/servlets/AnalyzedVideo.java]: My suggestion would be to not do any work inside the model (AnalyzedVideo). Instead, call this in `DataServlet` to get the sentiment score before setting it in the model.

**Review by jbiral — CHANGES_REQUESTED** [review]

File: src/main/webapp/script.js [hunk 7]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 function addThreads() {  // eslint-disable-line
-  const url = '/data';
+  const url = '/data1';
   fetch(url).then((response) => response.json()).then((threadInfoList) => {
     const threadList = document.getElementById('thread-container');
     threadList.innerHTML = '';
```
Comment (jonaner) [inline src/main/webapp/script.js:19]: Please remove the console log statement.

File: src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (shivanighanta) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: The getYoutubePost() method returns a list of all the youtube posts

Comment (jbiral) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: But can you iterate over the list of posts instead?

``` java
for (YoutubePost post : newPosts) {
  // Do something
}
```

**Review by jbiral — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (shivanighanta) [reply src/main/java/com/google/sps/servlets/DataServlet.java]: ok, got it!

File: src/main/java/com/google/sps/servlets/test/AnalyzeTest.java [hunk 8]
```diff
@@ -14,42 +14,41 @@
  * limitations under the License.
  */
```
Comment (jbiral) [inline src/main/java/com/google/sps/servlets/test/AnalyzeTest.java:16]: Could you move this file under: ```src/test/java/com/google/sps/servlets/AnalyzeTest.java```?

**Review by jbiral — APPROVED** [review]

**Pull request merged**